### ISSUE TITLE
Library cannot be used due to split package exported by dependencies

### ISSUE THREAD

**Issue by cowwoc** [kubernetes-client/java#3769]
**Describe the bug**
This library is incompatible with Java Modules. If you run `mvn dependency:tree` you will see:

```
[INFO]    +- io.kubernetes:client-java-api:jar:21.0.1:compile
[INFO]    |  +- javax.annotation:javax.annotation-api:jar:1.3.2:compile
[INFO]    |  +- jakarta.annotation:jakarta.annotation-api:jar:3.0.0:compile
[INFO]    |  +- com.google.code.findbugs:jsr305:jar:3.0.2:compile
```

1. The library depends on *both* `javax.annotation` and `jakarta.annotation`. I believe the latter is meant to replace the former.
2. Both `javax.annotation` and `jsr305` export the same package.

Anyone who tries declaring this library as a dependency will get this error:

`module X reads package javax.annotation from both jsr305 and java.annotation`

**Client Version**
21.0.1

**Java Version**
23

**To Reproduce**
Create an application that contains a `module-info.java` that contains the following lines:

```
requires io.kubernetes.client.java;
requires io.kubernetes.client.java.api;
```

**Expected behavior**
Ability to use from Java Modules. You'll need to replace all uses of `javax.annotation` by `jakarta.annotation` and ideally please find a replacement for JSR 305. See https://nipafx.dev/jsr-305-java-9/ for what a mess it is on Java 9+.

**Comment by cowwoc** [comment]
This issue is related to https://github.com/kubernetes-client/java/issues/2606

**Comment by brendandburns** [comment]
Happy to take PRs to improve this.

**Comment by venkat1701** [comment]
Hey, I can take up this issue if it is still open, it seems more like a dependency update with minor refactors. 
Thanks!

**Comment by cowwoc** [comment]
@venkat1701 Please do!

**Comment by venkat1701** [comment]
@cowwoc yeah, i've added exclusions for `javax.annotation` and 1. is solved. I am looking for solving the 2nd one
